starter-code/src/clue.js

```diff
@@ -1,94 +1,123 @@
+age:          36,
+image:        "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+occupation:   "Socialité"
+}
+var mrMustard ={
+first_name:   "Jack",
+last_name:    "Mustard",
+color:        "yellow",
+description:  "He is a former football player who tries to get by on his former glory",
+age:          62,
+image:        "https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg",
+occupation:   "Retired Football player"
+}
 // Weapons
 
-name: rope        ---   weight: 10
-name: knife       ---   weight: 8
-name: candlestick ---   weight: 2
-name: dumbbell    ---   weight: 30
-name: poison      ---   weight: 2
-name: axe         ---   weight: 15
-name: bat         ---   weight: 13
-name: trophy      ---   weight: 25
-name: pistol      ---   weight: 20
+var rope = { name:"rope" , weight:10};
+var knife = {name:"knife" ,weight: 8};
+var candlestick = {name: "candlestick",  weight: 2};
+var dumbell = {name: "dumbbell",  weight: 30};
+var poison = {name: "poison",   weight: 2};
+var axe = {name: "axe" ,  weight: 15};
+var bat={name: "bat",   weight: 13};
+var trophy={name: "trophy",   weight: 25};
+var pistol={name: "pistol",   weight: 20};
 
 // Rooms
-name: Dinning Room
-name: Conservatory
-name: Kitchen
-name: Study
-name: Library
-name: Billiard Room
-name: Lounge
-name: Ballroom
-name: Hall
-name: Spa
-name: Living Room
-name: Observatory
-name: Theater
-name: Guest House
-name: Patio
-*/
+var dinningRoom = {name: "Dinning Room"};
+ conservatory = {name: "Conservatory"};
+var kitchen = {name: "Kitchen"};
+var study={name: "Study"};
+var library={name: "Library"};
+var billiardRoom= {name: "Billiard Room"};
+var lounge={name: "Lounge"};
+var ballroom={name: "Ballroom"};
+var hall={name: "Hall"};
+var spa={name: "Spa"};
+var livinRoom={name: "Living Room"};
+var observatory={name: "Observatory"};
+var theater={name: "Theater"};
+var guestHouse={name: "Guest House"};
+var patio ={name: "Patio"};
+
 
 // Characters Collection
 var charactersArray = [];
+charactersArray.push(mrGreen,drOrchid,profPlum,missScarlet,mrsPeacock,mrMustard);
+
 
 // Rooms' Collection
 var roomsArray = [];
+roomsArray.push(dinningRoom,conservatory,kitchen,study,library,billiardRoom,lounge,ballroom,hall,spa,livinRoom,observatory,theater,guestHouse,patio);
+
 
 // Weapons Collection
 var weaponsArray = [];
+weaponsArray.push(rope, knife, candlestick, dumbell,poison,axe,bat,trophy,pistol);
+
+
+function randomSelector(array) {
+    return (array[Math.floor(Math.random()*array.length)])
+};
+console.log(randomSelector(charactersArray))
+
+function pickMistery () {
+    var randomCharacters = randomSelector (charactersArray);
+    var randomRooms = randomSelector (roomsArray);
+    var randomWeapons = randomSelector (weaponsArray);
+    var misteryArray = [randomCharacters,randomWeapons,randomRooms];
+    return misteryArray;
+}
+
+var misteryEnvelope = pickMistery();
+
+function revealMistery() {
```

Comment:
You have the right idea! You just needed to pass something to your function. I named it myMystery in the example bellow

```
function revealMstery(myMystery) {
  var result =
    myMystery[0].first_name +
    " " +
    myMystery[0].last_name +
    " killed Mr.Boddy using the " +
    myMystery[1].name +
    " in the " +
    myMystery[2].name +
    "!!!!";
  return result;
}
```

You also needed to swap around the order of the last two items in order for it to pass the test. They wanted the person then the weapon and then room.